### PR TITLE
MM-51456 - Update Overage Notices

### PR DESCRIPTION
**Pull request by ConorMacpherson**
#### Summary

Currently, in user overage notices, the call to actions for expanding seats redirects to the Portal. We need it to point to the in-product purchase modal for co-term expansion.

If the workspace is air-gapped, the customer is prompted to manage their account in the portal or by contacting their Customer Success Manager the same as they are currently.

These changes add additional states to both user overage notices (the system banner and warning in the invite modal). Both of these have a link to the license page with an `action` query param with a value of `show_expansion_modal`, which opens in a new tab, displaying the modal.

#### Ticket Link

[MM-51456](https://mattermost.atlassian.net/browse/MM-51456)

#### Screenshots

**Invite Modal With Users Overage Warning When Self-Hosted Expansion is Available**
image

**Users Overage Banner When Self-Hosted Expansion is Available**
image

**Self-Hosted Expansion Modal With Query Param**
image

#### Test Plan

**Setup**
1. Set up your workspace to have an overage user state (10% over licensed seats).

**Invite Modal**
1. Navigate to the invite modal in the right sidebar in mattermost.
2. Ensure the modal has a red warning banner stating "Your workspace user count has exceeded your paid license seat count by X seats", and "Purchase additional seats to remain compliant." (The "Purchase additional seats" part is a link which opens the self-hosted expansion modal when clicked.

**Overage Banner**
1. Naviagte to Channels.
2. Ensure there is an overage banner present that states: "Your workspace user count has exceeded your paid license seat count. Update your seat count to stay compliant.", and a button stating: "Purchase additional seats" (when clicked, it opens the self-hosted expansion modal).

#### Release Note

```release-note
NONE
```

[MM-51456]: https://mattermost.atlassian.net/browse/MM-51456?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webapp/channels/src/components/invitation_modal/overage_users_banner_notice/index.tsx [hunk 1]
```diff
@@ -83,7 +90,29 @@ const OverageUsersBannerNotice = () => {
     };
 
     let message;
-    if (!isGovSku) {
+
+    if (canSelfHostedExpand) {
+        message = (
+            <FormattedMessage
+                id='licensingPage.overageUsersBanner.selfHostedNoticeDescription'
+                defaultMessage={'<a>Purchase additional seats </a> to remain compliant.'}
```
Comment (neallred) [inline webapp/channels/src/components/invitation_modal/overage_users_banner_notice/index.tsx:98]: Nit: extra space in the link

**Review by neallred — APPROVED** [review]
LGTM, nothing blocking. Not sure if we want to get rid of the code that removes the open expansion modal query param.

**Comment by ConorMacpherson** [comment]
@nickmisasi friendly nudge for review.

**Review by nickmisasi — COMMENTED** [review]
One nit but otherwise LGTM! Sorry for the delay!

**Comment by ConorMacpherson** [comment]
@yasserfaraazkhan I had this PR open before the related changes were merged. Now that they are merged, do you want to test this before merge, or continue with post-merge for the whole feature? We don't actually need to go through the entire flow, just ensure that this opens the modal.

**Comment by yasserfaraazkhan** [comment]
> @yasserfaraazkhan I had this PR open before the related changes were merged. Now that they are merged, do you want to test this before merge, or continue with post-merge for the whole feature? We don't actually need to go through the entire flow, just ensure that this opens the modal.

Sure @ConorMacpherson , I'll try if the spinwick comes up, else will ping for post merge test

**Comment by ConorMacpherson** [comment]
> Sure @ConorMacpherson , I'll try if the spinwick comes up, else will ping for post merge test

Sounds good, thanks, Yasser! :)

**Comment by yasserfaraazkhan** [comment]
@ConorMacpherson spinwick fails to run. I made a note to do a test after merging

**Pull request merged**